Source/WebKit/Shared/WebFoundTextRange.h

```diff
@@ -55,10 +55,9 @@ struct WebFoundTextRangeHash {
 
 template<> struct HashTraits<WebKit::WebFoundTextRange> : GenericHashTraits<WebKit::WebFoundTextRange> {
     static WebKit::WebFoundTextRange emptyValue() { return { }; }
-    static WebKit::WebFoundTextRange deletedValue() { return { std::numeric_limits<uint64_t>::max(), 0, { }, 0 }; }
 
-    static void constructDeletedValue(WebKit::WebFoundTextRange& range) { range = deletedValue(); }
-    static bool isDeletedValue(const WebKit::WebFoundTextRange& range) { return range == deletedValue(); }
+    static void constructDeletedValue(WebKit::WebFoundTextRange& slot) { new (NotNull, &slot) WebKit::WebFoundTextRange { std::numeric_limits<uint64_t>::max(), 0, { HashTableDeletedValue }, 0 }; }
```

Comment:
Two other thoughts.

First, there is no need to overwrite all the data member of the WebFoundTextRange. We can just overwrite the one field used to hold a "deleted" flag. There is no need to touch of the other data members. Unless there is some subtle security reason we want to overwrite them, and I don’t think there is. So for one example of how this could work, we could just initialize the AtomString:
```
    new (NotNull, &slot.frameIdentifier) AtomString { HashTableDeletedValue };
```

Second, I am not sure it’s safe to compare this deleted value with `operator==` and if it’s not we have to set a boolean flag in the traits to tell the hash table implementation that.

Reply:
Updated to just initialize the AtomString.

I also updated `operator==` to account for the AtomString being a `HashTableDeletedValue`.